pkg/chain/ethereum/bonded_ecdsa_keep_handle.go

```diff
@@ -0,0 +1,335 @@
+package ethereum
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/keep-network/keep-common/pkg/chain/ethereum/ethutil"
+	"github.com/keep-network/keep-common/pkg/subscription"
+	"github.com/keep-network/keep-ecdsa/pkg/chain"
+	"github.com/keep-network/keep-ecdsa/pkg/chain/gen/contract"
+	"github.com/keep-network/keep-ecdsa/pkg/ecdsa"
+	"github.com/keep-network/keep-ecdsa/pkg/utils/byteutils"
+)
+
+type bondedEcdsaKeep struct {
+	handle   *ethereumChain
+	contract *contract.BondedECDSAKeep
+
+	contractAddress common.Address
+}
+
+// GetKeepWithID returns a handle to the BondedECDSAKeep with the provided id.
+func (bekm *bondedEcdsaKeepManager) GetKeepWithID(
```

Comment:
Wouldn't it be better to move this function to `bonded_ecdsa_keep_manager.go `?

Reply:
I tried something interesting across this PR, which is making the first function in each file answer the question “how do I get one of these?” This is related to something I'd like us to do in packages as well, which is making the first function in the package's eponymous Go file (e.g. `ethereum.go` for `pkg/ethereum`, `chain.go` for `pkg/chain`) answer the question “how do I get the main types this package exposes?”

Definitely open to doing it the other way though.

Reply:
Interesting idea. It sounds good to me if we move `func (bekm *bondedEcdsaKeepManager) GetKeepWithID(` to the mentioned eponymous `ethereum.go` file.

Reply:
Hmmm… Not sure if I follow, but also don't want to go too hard on this topic in this particular PR. It feels like a good fit for a smaller PR targeted at this specific refactoring.